### PR TITLE
Add vorticity as a diagnostic output

### PR DESCRIPTION
**Pull request by JFLemieux73**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 
   Added calculation of vorticity as a diagnostic for B and C grids. 
- [x] Developer(s): 
    @JFLemieux73 
- [x] Suggest PR reviewers from list in the column to the right.
    @eclare108213 
- [x] Please copy the PR test results link or provide a summary of testing completed below.
    I did the end-to-end procedure (base_suite):

385 measured results of 385 total results
385 of 385 tests PASSED
0 of 385 tests PENDING
0 of 385 tests MISSING data
0 of 385 tests FAILED

- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [x] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [ ] Please provide any additional information or relevant details below:

### CODE REVIEW

**Comment by JFLemieux73** [comment]
Because of the different location of u and v, the calculation of vorticity for the B and C grids are not exactly the same. The vorticity fields are however similar. Here is an example for the B-grid:

B_vort_2005-01-02-00000

and the C-grid:
C_vort_2005-01-02-00000

**Comment by JFLemieux73** [comment]
I forgot to mention that I am adding vorticity as an output in order to use Nils Hutter's linear kinematic features (LKFs) detection algorithm. Nils' tool uses vorticity to identify conjugate fault lines.

**Review by apcraig — CHANGES_REQUESTED** [review]
This looks reasonable.  Please add f_vort to set_nml.histall and set_nml.histdbg, set to same values as f_divu there.  Those namelist are used to test the history mechanisms more comprehensively.

Lots of work just to add one history field, thanks.

**Comment by apcraig** [comment]
I will test this on derecho when it's ready.  I can run the full io_suite which will cover binary, netcdf, and pio options.  Please let me know when you'd like me to do that.

**Comment by JFLemieux73** [comment]
Hi @apcraig,

I added f_vort to the two set_nml files. Please go ahead with the testing.

**Review by dabail10 — COMMENTED** [review]
Do we really need a vorticity variable? Is it used in any of the dynamic calculations? I would perhaps just do this as a temporary in ice_history.F90. This is the approach I've taken with other diagnostic variables that are not used in calculations.

**Comment by JFLemieux73** [comment]
This does not add a lot of computations because it is done after the EVP subcycling loop. I would keep the computation of vorticity together with divergence and shear (as proposed in this PR).  The shear strain rate invariant (a diagnostic) is currently calculated in the deformations subroutine. If we move the vorticity to ice_history.F90 then we should do the same for the shear. I suggest we keep both in the deformations subroutine. @eclare108213 what do you think about this?

**Comment by apcraig** [comment]
I ran a full test suite on derecho with intel, gnu, cray.  Results look good.  If things get updated again, I'm happy to retest.

**Review by apcraig — APPROVED** [review]

**Review by eclare108213 — APPROVED** [review]
I agree that vort belongs with divu and shear, conceptually.  It's not needed for the dynamics calculation but this is fine.  I also agree that this was a lot of work for a history variable!

**Comment by JFLemieux73** [comment]
Can we merge this PR?

File: cicecore/cicedyn/analysis/ice_history.F90
Comment (phil-blain) [inline cicecore/cicedyn/analysis/ice_history.F90:1329]: Should we specify at which location `vort` is defined ? I see shear, divu and strength do not mention it , but most other fields do.

File: cicecore/cicedyn/dynamics/ice_dyn_eap.F90
Comment (phil-blain) [inline cicecore/cicedyn/dynamics/ice_dyn_eap.F90:1375]: Here the computation of vorticity is the same as in `ice_dyn_shared::deformations`, right ?

If so, I would prefer we compute the vorticity in a separate subroutine, and call that subroutine from `ice_dyn_eap::stress_eap` and  `ice_dyn_shared::deformations`. This would reduce code duplication.

Comment (JFLemieux73) [reply cicecore/cicedyn/dynamics/ice_dyn_eap.F90:1375]: But then it is the same thing for divu and shear...the same computations are done in ice_dyn_eap and ice_dyn_shared. It has been like this for a long time...I say we keep it simple the way it is done right now.

**Comment by apcraig** [comment]
I just resolved some conflicts with the last PR and will run a short test suite to confirm no problems.  Once complete, I'll merge unless there is further discussion.

**Comment by apcraig** [comment]
I ran the quick_suite and io_suite on derecho with cray, gnu, intel.  All tests pass bit-for-bit.  Will merge now.

**Pull request merged**
